integration/src/main/java/scripts/testscripts/FolderLifecycle.java

```diff
@@ -248,6 +256,17 @@ protected void doUserJourney(TestUserSpecification testUsers, WorkspaceApi works
         HttpStatusCodes.STATUS_CODE_NOT_FOUND);
   }
 
+  private JobReport deleteFolderAndWaitForJobToFinish(Folder folderFoo)
+      throws ApiException, InterruptedException {
+    JobResult jobResult = folderWriterApi.deleteFolder(getWorkspaceId(), folderFoo.getId());
+    JobReport jobReport = jobResult.getJobReport();
+    while (ClientTestUtils.jobIsRunning(jobReport)) {
```

Comment:
You should have a result GETter for deleteFolder. I put a comment in the openapi yaml below. So you wouldn't wait on the jobs API. You would wait on the specific delete result.

Clearly you can do it as you have coded it, but I think it would be best to implement the result endpoint and test that here.

Reply:
done